src/Context/ContextDefinitionInterface.php

```diff
@@ -18,6 +18,14 @@
 interface ContextDefinitionInterface extends ContextDefinitionInterfaceCore {
 
   /**
+   * Constants for the context parameter restriction mode.
+   *
+   * @see ::getParameterRestriction()
+   */
+  const PARAMETER_RESTRICTION_INPUT = 'input';
```

Comment:
I think the wording needs work. it's no parameter any more but context in d8 - so the name does not fit any more. context restriction is not really specific, so what about "assignment restriction" - setAssignmentRestriction('input') ?